lib/fluent/config/types.rb

```diff
@@ -98,12 +98,24 @@ def self.regexp_value(str)
       end
       s
     }
-    INTEGER_TYPE = Proc.new { |val, opts| val.to_i }
-    FLOAT_TYPE = Proc.new { |val, opts| val.to_f }
-    SIZE_TYPE = Proc.new { |val, opts| Config.size_value(val) }
-    BOOL_TYPE = Proc.new { |val, opts| Config.bool_value(val) }
-    TIME_TYPE = Proc.new { |val, opts| Config.time_value(val) }
-    REGEXP_TYPE = Proc.new { |val, opts| Config.regexp_value(val) }
+    INTEGER_TYPE = Proc.new do |val, opts|
+      if opts[:strict]
+        Integer(val)
```

Comment:
Just dumping backtrace is not kindness for users.

How about dump error as follows?

```diff
diff --git a/lib/fluent/config/types.rb b/lib/fluent/config/types.rb
index 9c8b52fc..81cd7875 100644
--- a/lib/fluent/config/types.rb
+++ b/lib/fluent/config/types.rb
@@ -100,7 +100,11 @@ module Fluent
     }
     INTEGER_TYPE = Proc.new do |val, opts|
       if opts[:strict]
-        Integer(val)
+        begin
+          Integer(val)
+        rescue ArgumentError
+          raise ConfigError, "'#{val}' is not valid value for Integer."
+        end
       else
         val.to_i
       end
```

Reply:
> Just dumping backtrace is not kindness for users.

I think so too.
But I think dumping only `val` isn't still kindness. I want to dump also the value name for it.
I'm not sure how to do it at this moment but probably it can be realized by catching the error at a caller such as `SectionGenerator::generate`.

Reply:
I implemented it.